src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6337,7 +6543,88 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
+                    if (_currentInterpolatedStringHandlerContext.OutPlaceholder == -1)
+                    {
+                        throw ExceptionUtilities.Unreachable;
+                    }
+
+                    return new FlowCaptureReferenceOperation(_currentInterpolatedStringHandlerContext.OutPlaceholder, operation.Syntax, operation.Type, operation.GetConstantValue(), isInitialization: true);
+
+                case InterpolatedStringArgumentPlaceholderKind.CallsiteReceiver:
+                    if (_currentInterpolatedStringHandlerContext.HasReceiver && tryGetArgumentOrReceiver(-1) is IOperation receiverCapture)
+                    {
+                        Debug.Assert(receiverCapture is IFlowCaptureReferenceOperation);
+                        return OperationCloner.CloneOperation(receiverCapture);
+                    }
+                    else
+                    {
+                        return new InvalidOperation(ImmutableArray<IOperation>.Empty, semanticModel: null, operation.Syntax, operation.Type, operation.GetConstantValue(), isImplicit: true);
+                    }
+
+                case InterpolatedStringArgumentPlaceholderKind.CallsiteArgument:
+                    if (tryGetArgumentOrReceiver(operation.ArgumentIndex) is IOperation argumentCapture)
+                    {
+                        Debug.Assert(argumentCapture is IFlowCaptureReferenceOperation or IDiscardOperation);
+                        return OperationCloner.CloneOperation(argumentCapture);
+                    }
+                    else
+                    {
+                        return new InvalidOperation(ImmutableArray<IOperation>.Empty, semanticModel: null, operation.Syntax, operation.Type, operation.GetConstantValue(), isImplicit: true);
+                    }
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(operation.PlaceholderKind);
+            }
+
+            IOperation? tryGetArgumentOrReceiver(int argumentIndex)
+            {
+                int startingStackDepth = _currentInterpolatedStringHandlerContext.StartingStackDepth;
+                if (startingStackDepth >= _evalStack.Count)
+                {
+                    return null;
+                }
+
+                if (argumentIndex == -1)
+                {
+                    Debug.Assert(_currentInterpolatedStringHandlerContext.HasReceiver);
+
+                    // The receiver is always directly at the starting stack depth: there is no additional frame pushed on at
+                    // that point we need to be concerned with.
+
+                    IOperation? operation = _evalStack[startingStackDepth].operationOpt;
+                    Debug.Assert(operation != null);
+                    return operation;
+                }
+
+                // We need to walk forward until we hit a stack frame. If we hit a stack frame, then we're out of the allowable context
```

Comment:
>If we hit a stack frame, then we're out of the allowable context [](http://example.com/codeflow?start=71&length=64)

I think tt wold be more robust to not rely on this assumption. In order to achieve that we could simply record the stack depth at the time we enter ```VisitInterpolatedStringHandlerCreation``` in the context. That is the limit we don't want to go over.

Reply:
I don't think that would be more robust: in fact, I think it's more robust to ensure we stop on any new stack frame. A new stack frame means that we're in a new function call, specifically the the stack frame for the current `InterpolatedStringHandlerCreationOperation` being processed.

Reply:
The robust way is to not infer where to stop, but to know where to stop. If you think getting the stack depth in ```VisitInterpolatedStringHandlerCreation``` is still not strong enough, we can record position of each ```IInterpolatedStringHandlerCreationOperation``` in the argument list in the context and base logic on that. If you want to assert that there are no frames, that would be fine, but I don't think we should base our algorithm on that.